tests/llama_index/test_llama_index_model_export.py

```diff
@@ -344,3 +350,38 @@ def _get_query_embedding(self, *args, **kwargs: Any):
     response = loaded_model.predict("Spell llamaindex")
     assert isinstance(response, str)
     assert response != ""
+
+
+@pytest.mark.parametrize(
+    ("index_code_path", "vector_store_class"),
+    [
+        (
+            "tests/llama_index/sample_code/basic_vector_store.py",
+            SimpleVectorStore,
+        ),
+        (
+            "tests/llama_index/sample_code/external_vector_store.py",
+            QdrantVectorStore,
+        ),
+    ],
+)
+def test_save_load_index_as_code_optional_code_path(index_code_path, vector_store_class):
+    with mlflow.start_run():
+        model_info = mlflow.llama_index.log_model(
+            index=index_code_path,
+            engine_type="query",
+            artifact_path="model",
+            input_example="hi",
+        )
+
+    artifact_path = Path(_download_artifact_from_uri(model_info.model_uri))
+    assert os.path.exists(artifact_path / os.path.basename(index_code_path))
+    assert not os.path.exists(artifact_path / "index")
+    assert os.path.exists(artifact_path / "settings.json")
+
+    loaded_index = mlflow.llama_index.load_model(model_info.model_uri)
+    assert isinstance(loaded_index.vector_store, vector_store_class)
+
+    pyfunc_loaded_model = mlflow.pyfunc.load_model(model_info.model_uri)
+    assert isinstance(pyfunc_loaded_model.get_raw_model(), BaseQueryEngine)
+    assert pyfunc_loaded_model.predict("Spell llamaindex") != ""
```

Comment:
Is the return value not deterministic?

Reply:
It is deterministic because we use OAI mock, but the mock endpoint echos the input prompt, meaning that it includes internal prompt LlamaIndex uses. The prompt is not random but updated sometimes in their end.

Reply:
Replaced with `query in model.predict(query)`, which is still loose but slightly better.